### PR TITLE
ref(app-platform): change SentryAppWebhookError foreign keys to integer fields

### PR DESCRIPTION
**Pull request by nolachen**
We no longer need this model (SentryAppWebhookError ) since we are now storing the requests in Redis, not postgres

First step in deleting this model:
Turn all foreign key fields on SentryAppWebhookError into BoundedBigIntegerField so that Django doesn't expect any relations with this model

### CODE REVIEW

File: src/sentry/migrations/0013_auto_20191108_2104.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Adding indexes to large tables. These indexes should be created concurrently,
+    #   unfortunately we can't run migrations outside of a transaction until Django
+    #   1.10. So until then these should be run manually.
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
```
Comment (scefali) [inline src/sentry/migrations/0013_auto_20191108_2104.py]: @nolachen Should this be set to `True`?  I see that Matt recommended that.

Comment (nolachen) [reply src/sentry/migrations/0013_auto_20191108_2104.py]: @scefali hmm idk bc i feel like this isn't a risky migration

Comment (wedamija) [reply src/sentry/migrations/0013_auto_20191108_2104.py]: Yeah this migration is fine, it performs no database operations.

**Review by wedamija — APPROVED** [review]
Just following up from our discussion on why we should be cautious:

Since this model has a foreign key to `Organization`, it's possible that an `Organization` that links to a `SentryAppWebhookError` might be deleted. If that happened, then Django will automatically follow the links and attempt to delete related rows from `SentryAppWebhookError`. But if we'd deleted the table, then we'd fail to do that since the table doesn't exist. The same could happen for `SentryApp`. This would essentially prevent orgs/sentryapps from being deleted until the deploy had finished.

These errors are both pretty minor and probably wouldn't take down sentry, but are avoided by us breaking the deletion into two separate prs. This is mostly why we're so cautious around this, because it's hard to totally follow the consequences of what Django might do.

**Review by scefali — APPROVED** [review]

**Review by wedamija — APPROVED** [review]

File: src/sentry/models/sentryappwebhookerror.py [hunk 2]
```diff
@@ -3,19 +3,17 @@
...
 
     date_added = models.DateTimeField(db_index=True, default=timezone.now)
 
-    sentry_app = FlexibleForeignKey("sentry.SentryApp", related_name="webhook_errors")
+    sentry_app = BoundedBigIntegerField()
 
-    organization = FlexibleForeignKey(
-        "sentry.Organization", related_name="sentry_app_webhook_errors"
-    )
+    organization = BoundedBigIntegerField()
```
Comment (wedamija) [inline src/sentry/models/sentryappwebhookerror.py]: Learning more things about django migrations here. At the moment, this causes these fields to be renamed from `sentry_app_id` to `sentry_app`, which we definitely don't want.

To fix this, change the columns to `BoundedBigIntegerField(db_column="sentry_app_id")` and `BoundedBigIntegerField(db_column="organization_id")` and regenerate the migration.

**Review by wedamija — COMMENTED** [review]
I tested this with constraints with different names and it worked ok as well.
```
ALTER TABLE "sentry_sentryappwebhookerror" DROP CONSTRAINT "sent_organization_id_4fb7ee2f158ecff2_fk_sentry_organization_id";
ALTER TABLE sentry_sentryappwebhookerror ADD CONSTRAINT something_lol FOREIGN KEY (organization_id) REFERENCES sentry_organization(id);
```
Then the table looks like
`\d sentry_sentryappwebhookerror`
```
     Column      |           Type           |                                 Modifiers
-----------------+--------------------------+---------------------------------------------------------------------------
 id              | bigint                   | not null default nextval('sentry_sentryappwebhookerror_id_seq'::regclass)
 date_added      | timestamp with time zone | not null
 request_body    | text                     | not null
 request_headers | text                     | not null
 event_type      | character varying(64)    | not null
 webhook_url     | character varying(200)   | not null
 response_body   | text                     | not null
 response_code   | smallint                 |
 organization_id | bigint                   | not null
 sentry_app_id   | bigint                   | not null
Indexes:
    "sentry_sentryappwebhookerror_pkey" PRIMARY KEY, btree (id)
    "sentry_sentryappwebhooker_organization_id_4fb7ee2f158ecff2_uniq" btree (organization_id)
    "sentry_sentryappwebhookerro_sentry_app_id_70a927e7e9031158_uniq" btree (sentry_app_id)
    "sentry_sentryappwebhookerror_e39ce560" btree (date_added)
Check constraints:
    "sentry_sentryappwebhookerror_response_code_check" CHECK (response_code >= 0)
Foreign-key constraints:
    "sentry_se_sentry_app_id_70a927e7e9031158_fk_sentry_sentryapp_id" FOREIGN KEY (sentry_app_id) REFERENCES sentry_sentryapp(id) DEFERRABLE INITIALLY DEFERRED
    "something_lol" FOREIGN KEY (organization_id) REFERENCES sentry_organization(id)
```

`sentry django migrate sentry 0013`
Then the table looks like
`\d sentry_sentryappwebhookerror`
```
                                      Table "public.sentry_sentryappwebhookerror"
     Column      |           Type           |                                 Modifiers
-----------------+--------------------------+---------------------------------------------------------------------------
 id              | bigint                   | not null default nextval('sentry_sentryappwebhookerror_id_seq'::regclass)
 date_added      | timestamp with time zone | not null
 request_body    | text                     | not null
 request_headers | text                     | not null
 event_type      | character varying(64)    | not null
 webhook_url     | character varying(200)   | not null
 response_body   | text                     | not null
 response_code   | smallint                 |
 organization_id | bigint                   | not null
 sentry_app_id   | bigint                   | not null
Indexes:
    "sentry_sentryappwebhookerror_pkey" PRIMARY KEY, btree (id)
    "sentry_sentryappwebhookerror_e39ce560" btree (date_added)
Check constraints:
    "sentry_sentryappwebhookerror_response_code_check" CHECK (response_code >= 0)
```

**Review by wedamija — APPROVED** [review]
Looks good! Thanks for working through this with me.

For merging this, keep in mind that we need to merge/deploy each pr in order.

**Pull request merged**